### PR TITLE
EVG-17477 Add admin slack app name field and case on it in the slack help text 

### PR DESCRIPTION
**Pull request by malikchaya2**
[EVG-17477](https://jira.mongodb.org/browse/EVG-17477)

### Description 
This adds the "name" field to the admin slack config and only displays the second half of the help text if it's defined. 
I think I was the first to run gqlgen after the upgrade so it added a lot of comments. Please ignore them. 

### Testing 
Tested locally. 
image
image

### Spruce PR
https://github.com/evergreen-ci/spruce/pull/1476

### CODE REVIEW

File: graphql/host_resolver.go [hunk 1]
```diff
@@ -15,14 +15,17 @@ import (
 	"github.com/mongodb/grip/message"
 )
 
+// DistroID is the resolver for the distroId field.
```
Comment (khelif96) [inline graphql/host_resolver.go:18]: Did you run any special command to get these comments to generate or did you manually add them?

**Review by khelif96 — APPROVED** [review]

File: graphql/host_resolver.go (hunk 1 above)
Comment (malikchaya2) [reply graphql/host_resolver.go:18]: I think they all appeared after running `make gqlgen`. I didn't add them manually.

File: graphql/user_resolver.go [hunk 2]
```diff
@@ -37,11 +38,21 @@ func (r *userResolver) Patches(ctx context.Context, obj *restModel.APIDBUser, pa
...
 }
 
+// Permissions is the resolver for the permissions field.
 func (r *userResolver) Permissions(ctx context.Context, obj *restModel.APIDBUser) (*Permissions, error) {
 	return &Permissions{UserID: utility.FromStringPtr(obj.UserID)}, nil
 }
 
+// Target is the resolver for the target field.
+func (r *subscriberInputResolver) Target(ctx context.Context, obj *restModel.APISubscriber, data string) error {
+	panic(fmt.Errorf("not implemented: Target - target"))
```
Comment (Kimchelly) [inline graphql/user_resolver.go:48]: Is this supposed to panic?

Comment (malikchaya2) [reply graphql/user_resolver.go:48]: This was auto generated. @evergreen-ci/evg-ui how do you want to handle it?

Comment (ablack12) [reply graphql/user_resolver.go:48]: I'd strongly want to not panic fwiw

Comment (khelif96) [reply graphql/user_resolver.go:48]: This only gets generated if there's a mismatch between the go struct and the graphql schema is this possible there's a new target field on the subscription input field

Comment (malikchaya2) [reply graphql/user_resolver.go:48]: [subscription input schema](https://github.com/evergreen-ci/evergreen/blob/main/graphql/schema/types/user.graphql#L67)
[APISubscriber](https://github.com/evergreen-ci/evergreen/blob/main/rest/model/subscriber.go#L16)

the type for `target` are mismatched, but neither seem like new changes

Comment (Kimchelly) [reply graphql/user_resolver.go:48]: I guess the only questions I have here are, can this code ever be hit? If not, is it still necessary to keep as part of the schema, or can it be removed?

Comment (malikchaya2) [reply graphql/user_resolver.go:48]: We have this ticket open to address it: https://jira.mongodb.org/browse/EVG-18186

**Review by Kimchelly — APPROVED** [review]

**Review by ablack12 — APPROVED** [review]
Approving so this isn't blocked on me

**Pull request merged**